tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -570,6 +572,44 @@ public function testAlterTableScenario()
         }
     }
 
+
+    public function testAlterTableInNamespace()
+    {
+        if (! $this->_sm->getDatabasePlatform()->supportsSchemas()) {
+            $this->markTestSkipped('Schema definition is not supported by this platform.');
+        }
+
+        if (! in_array('testschema', $this->_sm->listNamespaceNames(), true)) {
+            $diff                  = new SchemaDiff();
+            $diff->newNamespaces[] = 'testschema';
+
+            foreach ($diff->toSql($this->_sm->getDatabasePlatform()) as $sql) {
+                $this->_conn->exec($sql);
+            }
+        }
+
+        $myTable = $this->createTestTable('testschema.my_table');
+        self::assertTrue(in_array('testschema.my_table', $this->_sm->listTableNames(), true));
+
+        $tableDetails = $this->_sm->listTableDetails('testschema.my_table');
+        self::assertTrue($tableDetails->hasColumn('id'));
+        self::assertTrue($tableDetails->hasColumn('test'));
+        self::assertTrue($tableDetails->hasColumn('foreign_key_test'));
+        self::assertCount(0, $tableDetails->getForeignKeys());
+        self::assertCount(1, $tableDetails->getIndexes());
+
+        $tableDiff                         = new TableDiff('testschema.my_table');
+        $tableDiff->fromTable              = $myTable;
+        $tableDiff->addedColumns['foo']    = new Column('foo', Type::getType('integer'));
+        $tableDiff->removedColumns['test'] = $myTable->getColumn('test');
+
+        $this->_sm->alterTable($tableDiff);
+
+        $tableDetails = $this->_sm->listTableDetails('alter_table');
```

Comment:
Should `alter_table` be `testschema.my_table` instead? Otherwise, the last assertion fails since there's no such table at all.

Reply:
OMG you are right! This assertion only did not fail in my test because one test before exactly this table was is created and modified in dbo.